java-monitoring/run_protoc.sh

```diff
@@ -0,0 +1,360 @@
+#!/usr/bin/env bash
+
+set -e
+
+LIBRARY_GEN_OUT=$(pwd)/../library_gen_out
+
+mkdir -p "${LIBRARY_GEN_OUT}"/com_google_protobuf
+mkdir -p "${LIBRARY_GEN_OUT}"/google/api
+mkdir -p "${LIBRARY_GEN_OUT}"/google/rpc
+mkdir -p "${LIBRARY_GEN_OUT}"/google/type
+mkdir -p "${LIBRARY_GEN_OUT}"/google/monitoring/v3
+
+##################### Section 0 #####################
+# prepare tooling
+#####################################################
+# proto files from googleapis repository
+REPO_ROOT=$(pwd)/..
+cd "${REPO_ROOT}"
+
+if [ ! -d googleapis ]; then
+  git clone https://github.com/googleapis/googleapis.git
+fi
+
+GOOGLEAPIS_ROOT=${REPO_ROOT}/googleapis
+cd "${GOOGLEAPIS_ROOT}"
+git checkout 53a0be29c4a95a1d3b4c0d3a7a2ac8b52af2a3c0
+# pull proto files and protoc from protobuf repository
+# maven central doesn't have proto files
+cd "${REPO_ROOT}"
+curl -LJ -o protobuf.zip https://github.com/protocolbuffers/protobuf/releases/download/v21.12/protoc-21.12-linux-x86_64.zip
+unzip -o -q protobuf.zip -d protobuf/
+cp -r protobuf/include/google "${GOOGLEAPIS_ROOT}"
+PROTOC_ROOT=${REPO_ROOT}/protobuf/bin
+echo "protoc version: $("${PROTOC_ROOT}"/protoc --version)"
+# pull protoc-gen-grpc-java plugin from maven central
+cd "${LIBRARY_GEN_OUT}"
+curl -LJ -o protoc-gen-grpc-java https://repo1.maven.org/maven2/io/grpc/protoc-gen-grpc-java/1.54.1/protoc-gen-grpc-java-1.54.1-linux-x86_64.exe
+chmod +x protoc-gen-grpc-java
+# gapic-generator-java
+curl -LJ -o gapic-generator-java.jar https://repo1.maven.org/maven2/com/google/api/gapic-generator-java/2.19.0/gapic-generator-java-2.19.0.jar
+##################### Section 1 #####################
+# generate grpc-google-cloud-monitoring-v3-java-srcs_pkg.tar.gz
+#####################################################
+cd "${GOOGLEAPIS_ROOT}"
+"${PROTOC_ROOT}"/protoc "--plugin=protoc-gen-rpc-plugin=${LIBRARY_GEN_OUT}/protoc-gen-grpc-java" \
+"--rpc-plugin_out=:${LIBRARY_GEN_OUT}/google/monitoring/v3/monitoring_java_grpc-proto-gensrc.jar" \
+google/monitoring/v3/alert.proto \
+google/monitoring/v3/alert_service.proto \
+google/monitoring/v3/common.proto \
+google/monitoring/v3/dropped_labels.proto \
+google/monitoring/v3/group.proto \
+google/monitoring/v3/group_service.proto \
+google/monitoring/v3/metric.proto \
+google/monitoring/v3/metric_service.proto \
+google/monitoring/v3/mutation_record.proto \
+google/monitoring/v3/notification.proto \
+google/monitoring/v3/notification_service.proto \
+google/monitoring/v3/query_service.proto \
+google/monitoring/v3/service.proto \
+google/monitoring/v3/service_service.proto \
+google/monitoring/v3/snooze.proto \
+google/monitoring/v3/snooze_service.proto \
+google/monitoring/v3/span_context.proto \
+google/monitoring/v3/uptime.proto \
+google/monitoring/v3/uptime_service.proto
```

Comment:
Can this be replaced with an `ls` on the directory?

Reply:
done